dp/generate_climos.py

```diff
@@ -373,10 +421,13 @@ def update_metadata_and_time_var(input_file, t_start, t_end, operation, climo_fi
 
 
         # Update cell_methods to reflect the operation being done to the data
+        # For this and for the frequency attribute, the "sum" operation is a mean:
+        # the between-years mean of the within-years or within-seasons sum.
         validate_operation(operation)
         cell_method_op = {
             'std': 'standard_deviation',
-            'mean': 'mean'
+            'mean': 'mean',
+            'sum': 'mean'
```

Comment:
So I guess at the end of the day this is a`mean` and not a `sum`, however I do think the metadata should specify the difference between the `mean` of some data and the `mean` of the `sum` of some data.  It _should_ reflect the fact that we've essentially done 2 sets of operations to the dataset, so something like: `time: sum over days time: mean over days`?  We should have a look at the [CF conventions](http://cfconventions.org/cf-conventions/cf-conventions.html#climatological-statistics) and figure out what would fit best.

Reply:
I think the cell method for non-climatology degree days going into this script is already `time: sum`, because that's the definition of degree days, sum the number of degrees above each threshold each day tp get the degree_days in a season, month, or year. Though perhaps I should make it say 'time: sum over days` instead?

So I don't know if we need to add anything extra to cell_methods besides the usual climatology stuff.

Reply:
@corviday , I am not sure what you mean here. CF Conventions is quite clear about what constitutes a valid climatology `cell_methods`. But I'm not sure what your comment implies about the `cell_methods` we should assign to these climatologies.

Reply:
I was answering @nikola-rados, but we both posted simultaneously, or near enough I didn't see your comment first. Probably it's confusing because it wasn't intended to address anything you said.

I said that going into the climatology operation, degree_days had a cell_methods of `time: sum` and thus came out of it with cell methods `time: sum time: mean over days`.

Reply:
Regrettably, that isn't one of the valid values for `cell_methods` required by [CF conventions, sec 7.4 Climatological Statistics](http://cfconventions.org/cf-conventions/cf-conventions.html#climatological-statistics). You have to specify the summation period. `sum within <interval>` is correct, where `<interval>` is the climatological sub-year interval. `sum over days` isn't right; it isn't even in the right form.

Reply:
Per face to face discussion with @rod-glover, cell methods changes probably need to be a separate issue and a separate PR. I won't be addressing them here.